compiler/AST/symbol.cpp

```diff
@@ -774,9 +774,16 @@ void VarSymbol::codegenDefC(bool global, bool isHeader) {
   // a variable can be codegen'd as static if it is global and neither
   // exported nor external.
   //
-  bool addExtern =  global && isHeader;
+  std::string str;
+
+  if(fIncrementalCompilation) {
+    bool addExtern =  global && isHeader;
+    str = (addExtern ? "extern " : "") + typestr + " " + cname;
+  } else {
+    bool isStatic =  global && !hasFlag(FLAG_EXPORT) && !hasFlag(FLAG_EXTERN);
+    str = (isStatic ? "static " : "") + typestr + " " + cname;
+  }
```

Comment:
The `+ typestr + " " + cname` operation is common to both branches of the if statement and so could be pulled to outside the end of it.  I wouldn't hold up the PR on this change (or even perform this action after the PR is merged), but it's something to be aware of when coding :)